expan/core/experiment.py

```diff
@@ -287,11 +287,65 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
         :return: boolean values indicating whether the row should be filtered
         :rtype: pd.Series
         """
-        method_table = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
-        na_replacement={'upper': -float_info.max        , 'lower':  float_info.max}
-
+        method_table   = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
+        na_replacement = {'upper': -float_info.max,         'lower':  float_info.max}
 
         for column in data[kpis].columns:
             threshold = np.percentile(data[column].fillna(na_replacement[threshold_type]), percentile)
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
+
+
+    def variant_split_is_consistent_with_expected_distribution(self, variant_column, weights, min_counts=5, alpha=0.05):
+        """ Tests the consistency of variant split with the hypothesized distribution.
+        
+        :param variant_column: variant column from the input data frame
+        :param weights: list of dicts with weights/split percentage per each variant 
+                        ([{'value':<variant>,'weight':<weight>}, ...])
```

Comment:
why not {'control': 0.5, 'treatment': 0.5} instead of that list of `value-weight` pairs?

Reply:
because there can be different number of variants and splits ratios (4 variants 25% each, for example).

Reply:
because there can be different number of treatments (1 control and 2 treatments, for example).

Reply:
what about `{'control': 0.5, 'treatment_1': 0.25, 'treatment_2': 0.25}`?

Reply:
I made it easy from expan service point of view: the endpoint returns 

"configuration": {
    "traffic_split_configuration": {
      "variants": [
        {
          "value": "Customization Modal disabled",
          "weight": 1000,
          "is_default": true,
          "image_url": "https://s3.eu-central-1.amazonaws.com/upload-variant-image-bucket/b8fee789-f250-495d-8065-8571e597ac77_bildschirmfoto_2018_04_17_um_18.32.12.png"
        },
        {
          "value": "Customization Modal enabled",
          "weight": 0,
          "is_default": false,
          "image_url": "https://s3.eu-central-1.amazonaws.com/upload-variant-image-bucket/49d35c6d-02c9-4426-9d37-c459fbab12d0_screen_shot_2018_04_17_at_20.16.10.png"
        }
      ],

I just want to grab the necessary json from the output.

Reply:
image_url ??
Please focus on lib APIs, make clear boundaries, don't overuse datastructures from the outer world, it may complicate maintenance in a long run.

Reply:
this was the example of the API return, I want to use those two fields (value, weight). How else can I get the traffic split?

Reply:
@igusher , @daryadedik , why are you discussing an endpoint? As far as I can see, this code neither consumes nor provides an endpoint. Such an endpoint wouldn't be in this repository anyway

Reply:
@aaron-mcdaid-zalando  by API I meant API which Expan lib provides. not some [rest] endpoints from any other  service

Reply:
@igusher I just misunderstood your comment. I will review my code when I come back.